### PR TITLE
Modify S2083(java): Improve the compliant sample

### PR DESCRIPTION
**Pull request by loris-s-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

**Review by gaetan-ferry-sonarsource — CHANGES_REQUESTED** [review]
There seem to be some confusion between Path and File.

File: rules/S2083/java/how-to-fix-it/java-se.adoc [hunk 1]
```diff
@@ -32,14 +32,14 @@ public class ExampleController
...
 {
     static private String targetDirectory = "/path/to/target/directory/";
+    Path targetPath = new File(targetDirectory).toPath().normalize();
 
     @GetMapping(value = "/delete")
     public void delete(@RequestParam("filename") String filename) throws IOException {
 
-        File file = new File(targetDirectory + filename);
-        String canonicalDestinationPath = file.getCanonicalPath();
+        File file = new Path(targetPath + filename);
```
[the remark this answers is not in the document]
Comment (loris-s-sonarsource) [reply rules/S2083/java/how-to-fix-it/java-se.adoc:40]:
```suggestion
        File file = new File(targetPath + filename);
```

File: rules/S2083/java/how-to-fix-it/java-se.adoc [hunk 2]
```diff
@@ -32,14 +32,14 @@ public class ExampleController
 public class ExampleController
 {
     static private String targetDirectory = "/path/to/target/directory/";
+    Path targetPath = new File(targetDirectory).toPath().normalize();
```
Comment (loris-s-sonarsource) [inline rules/S2083/java/how-to-fix-it/java-se.adoc:35]:
```suggestion
    static private String targetDirectory = "/path/to/target/directory/";
    Path targetPath                       = new File(targetDirectory).toPath().normalize();
```

File: rules/S2083/java/how-to-fix-it/java-se.adoc [hunk 3]
```diff
@@ -32,14 +32,14 @@ public class ExampleController
 public class ExampleController
 {
     static private String targetDirectory = "/path/to/target/directory/";
+    Path targetPath                       = new File(targetDirectory).toPath().normalize();
```
Comment (loris-s-sonarsource) [inline rules/S2083/java/how-to-fix-it/java-se.adoc:35]:
```suggestion
    private static String targetDirectory = "/path/to/target/directory/";
    private static Path targetPath        = new File(targetDirectory).toPath().normalize();
```

**Review by jamie-anderson-sonarsource — APPROVED** [review]
I haven't had a chance to verify that this works as expected, but the normalization of the path ensures that it can't be any worse than the previous solution.

File: rules/S2083/java/how-to-fix-it/java-se.adoc [hunk 4]
```diff
@@ -31,15 +31,15 @@ public class ExampleController
...
-    static private String targetDirectory = "/path/to/target/directory/";
+    private static String targetDirectory = "/path/to/target/directory/";
+    private static Path targetPath        = new File(targetDirectory).toPath().normalize();
 
     @GetMapping(value = "/delete")
     public void delete(@RequestParam("filename") String filename) throws IOException {
 
-        File file = new File(targetDirectory + filename);
-        String canonicalDestinationPath = file.getCanonicalPath();
+        File file = new File(targetPath + filename);
```
Comment (jamie-anderson-sonarsource) [inline rules/S2083/java/how-to-fix-it/java-se.adoc:40]: We may want to consider whether the following code is more reliable:
```suggestion
        File file = new File(targetPath.resolve(filename));
```

From what I can gather, `.resolve()` is the equivalent of a `path.join()` call in other languages.

**Pull request merged**